### PR TITLE
{Lab} Migrate `lab` command module to Microsoft Graph

### PR DESCRIPTION
**Pull request by jiasli**
**Related command**
`az lab`

**Description**
This PR depends on https://github.com/Azure/azure-cli/pull/28105.

`lab` command module directly use `azure-graphrbac` SDK to call AD Graph. 

This PR migrates `lab` command module to call Microsoft Graph to address https://github.com/Azure/azure-cli/issues/22174.

### CODE REVIEW

**Comment by yonzhan** [comment]
AD

**Comment by jiasli** [comment]
https://github.com/Azure/azure-cli/pull/29878 renamed `test_lab_gallery_vm_mgmt` to `test_lab_vm` and reenabled it. This causes CI to fail:

https://dev.azure.com/azclitools/public/_build/results?buildId=188942&view=logs&jobId=9ef86d90-5c36-539c-3501-167c2f35e379&j=9ef86d90-5c36-539c-3501-167c2f35e379&t=9f4da0ef-bb8b-52b5-2423-c7dae4abd8b8

```
except CannotOverwriteExistingCassetteException as ex:
>           raise AssertionError(ex)
E           AssertionError: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/azure-cli/azure/cli/command_modules/lab/tests/latest/recordings/test_lab_vm.yaml') in your current record mode ('once').
E           No match for the request (<Request (GET) [https://graph.microsoft.com/v1.0/me>](https://graph.microsoft.com/v1.0/me%3E)) was found.
E           Found 74 similar requests with 3 different matcher(s) :
E           
E           1 - (<Request (GET) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/labtest/providers/Microsoft.DevTestLab/labs/cliautomationlab?api-version=2018-09-15>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/labtest/providers/Microsoft.DevTestLab/labs/cliautomationlab?api-version=2018-09-15%3E).)
E           Matchers succeeded : ['method', 'scheme', 'port']
E           Matchers failed :
E           host - assertion failure :
E           graph.microsoft.com != management.azure.com
E           path - assertion failure :
E           /v1.0/me != /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/labtest/providers/Microsoft.DevTestLab/labs/cliautomationlab
E           _custom_request_query_matcher - assertion failure :
E           None
```

File: src/azure-cli/azure/cli/command_modules/lab/tests/latest/recordings/test_lab_vm.yaml [hunk 1]
```diff
@@ -2339,11 +2339,10 @@ interactions:
       accept-language:
       - en-US
     method: GET
-    uri: https://graph.windows.net/00000000-0000-0000-0000-000000000000/me?api-version=1.6
+    uri: https://graph.microsoft.com/v1.0/me
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/lab/tests/latest/recordings/test_lab_vm.yaml:2342]: I don't know why `test_lab_vm` is marked as `record_only`:

https://github.com/Azure/azure-cli/blob/fd5779f74f78aad16d56c67a1644d4299a5904ff/src/azure-cli/azure/cli/command_modules/lab/tests/latest/test_lab.py#L19-L20

so I manually edited the recording file.

File: src/azure-cli/azure/cli/command_modules/lab/tests/latest/recordings/test_lab_vm.yaml [hunk 2]
```diff
@@ -2739,11 +2738,10 @@ interactions:
...
+      string: '{"objectId":"d44a2991-98c6-47c3-b59b-2b30d72cfcc2"}'
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/lab/tests/latest/recordings/test_lab_vm.yaml:2744]:
```suggestion
      string: '{"id":"d44a2991-98c6-47c3-b59b-2b30d72cfcc2"}'
```
Test failed with 
```
graph_client = <azure.cli.command_modules.role._msgrpah._graph_client.GraphClient object at 0x7f3f2bab9f50>

    def _get_current_user_object_id(graph_client):
        # Calling this API with a service principal raises GraphError:
        #   /me request is only valid with delegated authentication flow.
        current_user = graph_client.signed_in_user_get()
>       return current_user['id']
E       KeyError: 'id'
```

**Review by evelyn-ys — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Pull request merged**